expression/builtin_cast_vec.go

```diff
@@ -517,11 +518,48 @@ func (b *builtinCastIntAsDecimalSig) vecEvalDecimal(input *chunk.Chunk, result *
 }
 
 func (b *builtinCastIntAsJSONSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCastIntAsJSONSig) vecEvalJSON(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalInt(b.ctx, input, buf); err != nil {
+		return err
+	}
+	nums := buf.Int64s()
+	result.ReserveJSON(n)
+	if mysql.HasIsBooleanFlag(b.args[0].GetType().Flag) {
```

Comment:
This may have better performance.
``` go
for i := 0; i < n; i++{
    if{
    } else if{}
}
```

Reply:
I'm not sure. I think it will caculate less `if` when putting `if mysql.HasXXXFlag(...)` outside of the `for` loop. When putting the `if` in the `for` loop, for every `i` it will caculate a `if`. However, the CPU may take some optimization in this condition. It's hard to measure which method is faster.

Reply:
OK, I got you. I think you're right.